tests/raftstore/test_multi.rs

```diff
@@ -121,6 +125,56 @@ fn test_multi_lost_majority<T: Simulator>(cluster: &mut Cluster<T>, count: usize
 
 }
 
+fn test_multi_random_restart<T: Simulator>(cluster: &mut Cluster<T>,
+                                           node_count: usize,
+                                           restart_count: u32) {
+    cluster.bootstrap_region().expect("");
+    cluster.start();
+
+    let (key, value) = (b"a1", b"v1");
+
+    assert_eq!(cluster.get(key), None);
+    cluster.put(key, value);
+    assert_eq!(cluster.get(key), Some(value.to_vec()));
+
+    let mut rng = rand::thread_rng();
+    for _ in 1..restart_count {
+        let id = 1 + rng.gen_range(0, node_count as u64);
+        cluster.stop_node(id);
+        cluster.run_node(id);
+        wait_until_node_online(cluster, id);
+        assert_eq!(cluster.get(key), Some(value.to_vec()));
+
+        // verify whether data is actually being replicated
+        must_get_equal(&cluster.get_engine(id), key, value);
+
+        let (k, v) = (b"a2", b"v2");
+        assert_eq!(cluster.get(k), None);
+        cluster.put(k, v);
+        assert_eq!(cluster.get(k), Some(v.to_vec()));
+        cluster.delete(k);
+        assert_eq!(cluster.get(k), None);
+    }
+}
+
+fn wait_until_node_online<T: Simulator>(cluster: &mut Cluster<T>, node_id: u64) {
+    loop {
```

Comment:
Call must_get_equal directly can't fulfill the task here. 
Unless I delete the key in the engine first, otherwise must_get_equal can't ensure raft leader is elected out.
Maybe I should add a retry timeout, tomorrow, if you persist~